File: C++/53_MaximumSubArray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int allSum = 0;
        int maxSum = nums[0];
        for (int i : nums){
            allSum += i;
            maxSum = std::max(allSum, maxSum);
            if (allSum < 0) {
                allSum = 0;
            }
        }
        return maxSum;
    }
};

File: 53_MaximumSubArray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        allSum = 0 
        maxSum = nums[0]
        for i in nums:
            allSum += i
            maxSum = max(allSum, maxSum)
            if allSum < 0:
                allSum = 0
        return maxSum